analyses/01_sample_distribution_analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

```diff
@@ -0,0 +1,285 @@
+
+
+# File path to results directory
+results_dir <-
+  file.path(root_dir, "results")
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+
+
+# File path to plots directory
+plots_dir <-
+  file.path(root_dir, "plots")
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+
+# Outputs
+output_PBTA_filter <- file.path(results_dir, "data_by_cohort_PBTA_filter.tsv")
+output_PBTA_filter_count <- file.path(results_dir, "data_by_cohort_PBTA_filter_count.tsv")
+```
+
+
+## Load data
+```{r load-inputs-please-wait}
+# Inputs
+## Read in metadata
+histologies <- readr::read_tsv(input_metadata, guess_max = 100000, show_col_types = FALSE)
+# dim(data)
+
+# Read table with colors per histology
+cols <- readr::read_tsv(input_cols, guess_max = 100000, show_col_types = FALSE) # Load plotting colors
+
+# Add colors palette to the histologies file
+hist_cols <- histologies %>% left_join(cols, by=c('cancer_group','broad_histology'))  #to merge tables
+# This is the tidyverse method. Try not to name variables the same as a function's name - that may cause coding errors downstream.
+
+
+# to discard when add the col palette for PBTA cohort
+mypalette <- brewer.pal(12,"Paired") # colorblindFriendly = TRUE
+col5 <- colorRampPalette(mypalette)(5)
+col9 <- colorRampPalette(mypalette)(9)
+
+```
+
+# Process Data
+
+All tumor samples have matched normal samples. 
+We need to filter normal samples out for further analysis. 
+We are using "pathology_diagnosis" instead of "sample_type". This way we discard normal samples (as shown in "sample_type") in addition to undiagnosed tumor samples.
+Also, we will discard Derived Cell Line, PDX from "composition".
+
+
+```{r cohort PBTA, echo=TRUE}
+data <- hist_cols  %>%
+   filter(cohort == "PBTA",
+          !is.na(pathology_diagnosis),
+          !composition %in% c("Derived Cell Line", "PDX")) %>% 
+         # !is.na(cancer_group) # Discard NAs in cancer type
+  mutate(tumor_descriptor = ifelse(sample_id == "C4106355-Tumor" & tumor_descriptor == "Not Reported", "Recurrence", 
+                                   ifelse(sample_id == "C4113612-Tumor" & tumor_descriptor == "Initial CNS Tumor", "Recurrence", 
+                                          tumor_descriptor)))
+
+# Note
+# The following samples (sample_id) are marked as Not Reported tumor_descriptor: 7316UP-3705, C4106355-Tumor, and C4113612-Tumor
+# Jo Lynne found the following status for 2/3 in the data warehouse:
+# C4106355-Tumor = Recurrence
+# C4113612-Tumor = Initial CNS Tumor
+# 7316UP-3705 = Not Reported
+# manually annotate them here and these will be updated in the v13 - then discard that part of the code
+
+
+data$tumor_descriptor <- ifelse(data$tumor_descriptor=="Progressive Disease Post-Mortem", "Deceased",
+                                 ifelse(data$tumor_descriptor=="Initial CNS Tumor", "Diagnosis",
+                                 ifelse(data$tumor_descriptor=="Primary Tumor", "Diagnosis", data$tumor_descriptor)))
```

Comment:
I think it would be easier if you add this logic to the above mutate function - to keep consistency with using tidyverse above over base here, and then you can also combine the two instances of making the tumor "Diagnosis" (ifelse tumor_descriptor %in% c("Initial CNS Tumor", "Primary Tumor")....